publisher-sdk/src/androidTest/java/com/criteo/publisher/EpochClockTest.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.criteo.publisher
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class EpochClockTest {
+
+  @Test
+  fun getApplicationTimeInMillis_GivenSdkInitialized_ApplicationTimeIsStarted() {
+    CriteoUtil.givenInitializedCriteo()
+
+    Thread.sleep(500)
```

Comment:
Sleeping in a test is a code smell when the sleep is used as a high bound of the an execution time (like when sync a test via sleep): with a variable execution time, then the test is flaky, the bound is made bigger, ...

In this case the sleep is there to literally check that time has spent. So it is used a lower bound, and even if the execution time is variable, it would never break the test, except if the property "time has spent" is violated.

Reply:
This is what clock mocks are for: not having to wait in tests. This why I am suggesting passing a mock `Clock` to a class whose responsibility would be to calculate the session time. There is no value in actually sleeping if we are able to manually modify the `Clock` through a mock.